src/Bicep.Core/Semantics/SemanticModel.cs

```diff
@@ -186,12 +188,35 @@ public IEnumerable<IDiagnostic> GetAllDiagnostics()
             return AssembleDiagnostics();
         }
 
-        private IEnumerable<IDiagnostic> AssembleDiagnostics()
+        private IReadOnlyList<IDiagnostic> AssembleDiagnostics()
         {
-            return GetParseDiagnostics()
-            .Concat(GetSemanticDiagnostics())
-            .Concat(GetAnalyzerDiagnostics())
-            .OrderBy(diag => diag.Span.Position);
+            var diagnostics = GetParseDiagnostics()
+                .Concat(GetSemanticDiagnostics())
+                .Concat(GetAnalyzerDiagnostics())
+                .OrderBy(diag => diag.Span.Position);
+            var filteredDiagnostics = new List<IDiagnostic>();
+
+            var disabledDiagnosticsCache = SourceFile.DisabledDiagnosticsCache;
+            foreach (IDiagnostic diagnostic in diagnostics)
+            {
+                (int diagnosticLine, _) = TextCoordinateConverter.GetPosition(SourceFile.LineStarts, diagnostic.Span.Position);
+
+                if (diagnosticLine == 0 || (diagnostic.Level == DiagnosticLevel.Error && diagnostic.GetType() != typeof(AnalyzerDiagnostic)))
```

Comment:
We've been using the C# pattern matching syntax - e.g.:
```csharp
diagnostic is AnalyzerDiagnostic
```

One benefit is this also works for classes which extend `AnalyzerDiagnostic` (in case that's something we add in the future).

Separately, I think it would be great to put this logic somewhere reusable (`CanBeSuppressed()` extension method on `IDiagnostic`?) so that we can ensure this check stays in sync with the check in `CodeActionHandler` if other people make changes to it.

Reply:
Sure, updated here:
https://github.com/Azure/bicep/commit/2015484d7222cf4c83196b921d89a017cbc22a04